_posts/2017-04-24-solarized.md

```diff
@@ -0,0 +1,545 @@
+---
+layout: post
+title:  "Solarized Theme"
+date:   2017-04-24 17:20:00 +0900
+categories: solarized archlinux i3 termite tmux zsh vim
+---
+
+In this post, I will go over my configuration to set the Solarized light/dark
+themes in linux. In addition, I'll share my tricks to easily toggle between these
+2 themes across all applications.
+
+Below are the applications I will refer to:
+
+* [Arch Linux](https://www.archlinux.org/): Lightweight rolling-release Linux distribution.
+* [I3](https://i3wm.org/): Tiling window manager.
+* [Termite](https://github.com/thestinger/termite): Keyboard-centric VTE-based
+  terminal.
+* [Tmux](https://tmux.github.io/): Terminal multiplexer to easily switch between
+  programs in a single terminal.
+* [NeoVim](https://github.com/neovim/neovim): Vim fork, text editor.
+
+# [](#solarized)Solarized
+
+> [Solarized](http://ethanschoonover.com/solarized) is a color palette designed to
+> have very readable files in both light and dark modes, using only 16 colors. The
+> colors were chosen from the CIELAB color space, which smartly arranges colors
+> visible to the human eye.
+
+![solarized palette](/assets/images/solarized-palette.png)
+
+There are many other interesting color schemes out there, like
+[gruvbox](https://github.com/morhetz/gruvbox), but I won't discuss the
+differences between each palette in this article.
+
+I have been using Solarized for several years now, and a recurrent issue was
+setting up my system (everytime I'd change laptop) to homogeneize colors across
+applications).
+
+While it would be nice to see applications reuse default colors, they
+unfortunately define their own set of colors, and we therefore need to copy the
+solarized color palette in each of these configuration files.
+
+Imagine you are running Vim within Tmux, both setup with solarized dark colors,
+and want to switch to the light theme because you're working outside or near a
+window around noon. You might already have a vim binding allowing you to toggle
+the colorscheme mode within your current session, but then you'll be struck by
+the high contrast between the tmux dark colors shown in other panes and the vim
+light colors in your current pane. Moreover, if you'd like to reopen a new vim
+session, you'll be back to the default theme, which is rather annoying.
+
+![Solarized contrast in Vim and
+Tmux](/assets/images/2017-04-24-solarized/solarized_vim_tmux_contrast.png)
+
+I will discuss a solution to bind all keystrokes from all applications in a
+single place, and smoothly transition between each color mode.
```

Comment:
Here's how I smoothly transition between light and dark mode for all applications using a shortcut.